### PR TITLE
ensure all raw sensor input orientations are normalized even if messages are not

### PR DESCRIPTION
**Pull request by SteveMacenski**
I've ran into issues where IMU data may or may not always be correctly normalized for TF's liking. Robot Pose EKF deals with this issue gracefully via tf::quaternionMsgToTF which will normalize the orientation quaternion before throwing it into the EKF. I have added the normalization call between Quaternion data unpack and utilization.

### CODE REVIEW

**Comment by ayrton04** [comment]
Thanks for the PR!

Is it possible for us to detect this condition, and spit one a `ROS_WARN_ONCE` message before we correct it? I feel like sensor data ought to conform to standards, and while I'm happy to correct it within the package, it could speak to other issues with the sensor or sensor drive that the user ought to know about and address.

**Comment by SteveMacenski** [comment]
That's a very thoughtful suggestion. I will of course add it this evening. 

Separate of this PR, I was curious if you had any thoughts on [this ticket in ros answers](https://answers.ros.org/question/272226/robot_localization-ouputs-nans-when-imuodom-messages-not-naning-works-in-robotposeekf/).  I found this minor shortcoming in robot_localization while trying to fight this issue.

**Comment by ayrton04** [comment]
I've answered your question. Let me know how it behaves once you correct the IMU messages.

File: src/ros_filter.cpp [hunk 1]
```diff
@@ -2318,6 +2318,11 @@ namespace RobotLocalization
         else
         {
           tf2::fromMsg(msg->orientation, curAttitude);
+          if (fabs(curAttitude.length() - 1.0) > 0.01)
```
Comment (ayrton04) [inline src/ros_filter.cpp:2321]: Do we know what kind of tolerance for normalization we require? The PR job passed, so this isn't affecting any of the bags in the test data set, which means that either their quaternion lengths were within this tolerance, or the normalization doesn't have a strong effect (I'm assuming the former).

**Comment by SteveMacenski** [comment]
Thanks for the feedback. I am looking into it next week; if it turns out to be the case, I will try to build infrastructure for robot_localization to error to the user the cause if its inverting. I've seen a number of ros answers tickets that could benefit from it as well.

I added the ros warn once before normalizations. Let me know if you want me to do anything else before merging the PR.

**Comment by ayrton04** [comment]
I am happy to merge this PR, but I am not sure how I will feel about a PR that checks for or corrects for bad covariance matrices in the input. If I'm just checking diagonal values for being non-negative or zero, that's one thing, but checking the determinant of every measurement covariance matrix is going to have an impact on performance. At some point, I think the onus needs to be on the user to make sure that the data that they are feeding to the filter is valid.

**Comment by ayrton04** [comment]
Thanks again for the PR!

**Comment by SteveMacenski** [comment]
My thought would have been to check a value in the covariance matrix before inversion and after and see if it turns from a number to a NaN and spit out an error indicating to the user where it happened and why. At least that way people can make up their own decision as to what to do. Is that more in line with something you would accept?

**Comment by ayrton04** [comment]
That could work. It's this line:

https://github.com/cra-ros-pkg/robot_localization/blob/kinetic-devel/src/ekf.cpp#L160

**Pull request merged**